core/src/main/java/com/coderanch/blackjack/Hand.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (C) 2018 Coderanch.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.coderanch.blackjack;
+
+import java.util.*;
+
+import static com.coderanch.util.require.Require.requireThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+/**
+ * A hand in a game of Blackjack.
+ */
+final class Hand {
+
+    /**
+     * Highest score in Blackjack.
+     */
+    private static final int MAX_LEGAL_SCORE = 21;
+
+    /**
+     * The cards in the hand.
+     */
+    private final List<Card> cards;
+
+    /**
+     * Creates a new hand.
+     *
+     * @param card  first card dealt.
+     * @param card2 second card dealt
+     * @throws NullPointerException if either {@code card} or {@code card2} is {@code null}.
```

Comment:
Throw `IllegalArgumentException` and use `requireThat` to validate the arguments.

Reply:
It doesn't compile if you put `requireThat` before calling `this()`.  I auto commited the change you made but removed it.   Am I missing something?

Reply:
It doesn't compile if you put requireThat before calling this(). I auto commited the change you made but removed it. Am I missing something?

Reply:
```java
    /**
     * Creates a new hand.
     *
     * @param firstCard  the first card dealt.
     * @param secondCard the second card dealt
     * @throws IllegalArgumentException if either {@code firstCard} or {@code secondCard} is {@code null}.
     */
    Hand(Card firstCard, Card secondCard) {
        this(List.of(
            requireThat("firstCard",  firstCard,  is(notNullValue())),
            requireThat("secondCard", secondCard, is(notNullValue()))
        ));
    }
```